### ISSUE TITLE
IDEA import fails with `buildSrc` project and `org.gradle.java.home` specified

### ISSUE THREAD

**Issue by bigdaz** [gradle/gradle#8353]
Given a simple project with:
- a `buildSrc` project
- a top-level `gradle.properties` setting `org.gradle.java.home` to a JDK 9+

When attempting to import/sync in Intellij IDEA, the daemon process cannot be started.

### Expected Behavior
Build will sync/import without error.

### Current Behavior
Sync fails after initial "CONFIGURE SUCCESSFUL:
```
> Task :buildSrc:compileJava UP-TO-DATE
> Task :buildSrc:compileGroovy NO-SOURCE
> Task :buildSrc:processResources NO-SOURCE
> Task :buildSrc:classes UP-TO-DATE
> Task :buildSrc:jar UP-TO-DATE
> Task :buildSrc:assemble UP-TO-DATE
> Task :buildSrc:compileTestJava NO-SOURCE
> Task :buildSrc:compileTestGroovy NO-SOURCE
> Task :buildSrc:processTestResources NO-SOURCE
> Task :buildSrc:testClasses UP-TO-DATE
> Task :buildSrc:test NO-SOURCE
> Task :buildSrc:check UP-TO-DATE
> Task :buildSrc:build UP-TO-DATE

CONFIGURE SUCCESSFUL in 0s
Unable to start the daemon process.
This problem might be caused by incorrect configuration of the daemon.
For example, an unrecognized jvm option is used.
Please refer to the user guide chapter on the daemon at https://docs.gradle.org/5.1.1/userguide/gradle_daemon.html
Please read the following process output to find out more:
-----------------------
Error: Could not find or load main class java.base.java.lang=ALL-UNNAMED
Caused by: java.lang.ClassNotFoundException: java.base.java.lang=ALL-UNNAMED

```

![image](https://user-images.githubusercontent.com/179734/51857396-5fb90580-22ef-11e9-818b-919612ae0509.png)

### Context
This appears to be an issue when IDEA attempts to separately get the model for the `buildSrc` build.

Symptoms are similar to #7045 but I suspect the underlying cause is different. This is an issue starting the Gradle daemon itself, whereas #7045 is related to starting a Groovy compiler daemon.

Possibly related issues:
https://youtrack.jetbrains.com/issue/IDEA-202975
https://youtrack.jetbrains.com/issue/IDEA-203429

### Steps to Reproduce (for bugs)

Clone https://github.com/baron1405/gradle7045 and follow instructions in the README.

Thanks to @baron1405 for this reproducer.

**Comment by bigdaz** [comment]
I believe I've tracked this down to an issue in the way IntelliJ IDEA treats `buildSrc` and JVM arguments:
- IDEA requests a TAPI model for the main build, this spawns a daemon process
- Gradle adds some `--add-opens` arguments when starting the daemon process, to avoid gratuitous warnings when processing Groovy code. This is represented as a list like `[--add-opens, java.base/java.util=ALL-UNNAMED, --add-opens, java.base/java.lang=ALL-UNNAMED]`.
- IDEA then requests a TAPI model for the `buildSrc` build:
   - To avoid spawning a new daemon process, IDEA reads the JVM args for the original build and passes them in as arguments for the `buildSrc` request. [These JVM args are read as a list from the Gradle `BuildEnvironment`](https://github.com/JetBrains/intellij-community/blob/9b98aa709817c62e2061864102bd7c6ed9156826/plugins/gradle/src/org/jetbrains/plugins/gradle/service/project/GradleBuildSrcProjectsResolver.java#L101-L107), with duplicate `--add-opens` retained.
   - [These JVM arguments are then added to a **`Set`**](https://github.com/JetBrains/intellij-community/blob/4999f5293e4307870020f1d0d672a3d35a52f22d/platform/external-system-api/src/com/intellij/openapi/externalSystem/model/settings/ExternalSystemExecutionSettings.java#L80-L83), which loses the duplicate `--add-opens` arguments from the list. The arguments look like `[--add-opens, java.base/java.util=ALL-UNNAMED, java.base/java.lang=ALL-UNNAMED]`
   - If a new daemon is required for `buildSrc`, then spawning this daemon will fail because the VM args are wrong.

So without `org.gradle.java.home`, we don't try to spawn a new process for the `buildSrc` request, and this bug is never exposed. Somehow, adding `org.gradle.java.home` makes the original daemon process incompatible, forcing a new daemon process, and triggering this bug.

In the end, I think this needs to be fixed on the IDEA side, either by removing the `--add-opens` pairs from the JVM args, or preserving the duplicates.

**Comment by baron1405** [comment]
... and that explains why this bug looks like #7045. Both gradle, in the case of #7045, and IDEA, in this case, are ultimately using a Set to hold the command line arguments. Good sleuthing @bigdaz! Can you open an issue on JetBrains' Youtrack for this.

**Comment by baron1405** [comment]
Thanks @sormuras, I added to the Jebrains issue a link to my sample project.

**Comment by bigdaz** [comment]
@sormuras @baron1405 Yep, I reached out directly to JetBrains on this, and given the change in status of the existing issue I don't believe we need to file a new issue on their issue tracker.

**Comment by big-guy** [comment]
@bigdaz should we do something on our side to make the failure message better? 

e.g., 
> Unable to start the daemon process with arguments: 
> $args

**Comment by bigdaz** [comment]
@big-guy I had to make exactly that change in order to diagnose the issue locally, so I think this would be a worthwhile change to make. I can probably find some time for that over the coming week.

**Comment by baron1405** [comment]
I have verified that this has been fixed in IntelliJ IDEA 2018.3.5. This issue can be closed.
